api-server/src/server/boot/challenge.js

```diff
@@ -711,15 +711,21 @@ function createMsTrophyChallengeCompleted(app) {
       );
 
       if (!challenge) {
-        throw new Error('Challenge not found');
+        throw new Error('flash.ms.trophy.err-2');
       }
 
       const { msTrophyId = '' } = challenge;
       const msTrophyApiUrl = `https://learn.microsoft.com/api/gamestatus/achievements/${msTrophyId}?username=${msUsername}&locale=en-us`;
       const msApiRes = await fetch(msTrophyApiUrl);
 
       if (!msApiRes.ok) {
-        throw new Error('Unable to validate trophy');
+        return res.status(500).json({
```

Comment:
```suggestion
        return res.status(404).json({
```
I don't think we should be uisng 500 here, since we know what's wrong and we know how to handle it. 404 seems like the best since the server looked for evidence they'd completed the trophy and didn't find it.